src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -0,0 +1,264 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.ReliableConnection;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+
+namespace Microsoft.SqlTools.ServiceLayer.FileBrowser
+{
+    /// <summary>
+    /// Main class for file browser service
+    /// </summary>
+    public sealed class FileBrowserService
+    {
+        private static readonly Lazy<FileBrowserService> LazyInstance = new Lazy<FileBrowserService>(() => new FileBrowserService());
+        public static FileBrowserService Instance => LazyInstance.Value;
+
+        // Cache file browser operations for expanding node request
+        private Dictionary<string, FileBrowserOperation> ownerToFileBrowserMap = new Dictionary<string, FileBrowserOperation>();
+        private ConnectionService connectionService = null;
+
+        private Dictionary<string, ValidatePathsCallback> validatePathsCallbackMap { get; set; }
```

Comment:
>validatePathsCallbackMap [](start = 58, length = 24)

properties should use PascalCasing where the first letter is Upper-Case...fields are camelCased..this is per standard .Net coding convention guidelines.

Reply:
..not sure why this is a property instead of field?  and why it's initialized in constructor instead of inline here, or lazily if need be.  If there isn't a reason please refactor.

---
In reply to: [137413210](https://github.com/Microsoft/sqltoolsservice/pull/448#discussion_r137413210) [](ancestors = 137413210)